src/AppInstallerCommonCore/Manifest/ManifestYamlPopulator.cpp

```diff
@@ -472,9 +1766,62 @@ namespace AppInstaller::Manifest
             {
                 std::vector<FieldProcessInfo> fields_v1_2 =
                 {
-                    { "PurchaseUrl", [this](const YAML::Node& value)->ValidationErrors { m_p_localization->Add<Localization::PurchaseUrl>(value.as<std::string>()); return {}; } },
-                    { "InstallationNotes", [this](const YAML::Node& value)->ValidationErrors { m_p_localization->Add<Localization::InstallationNotes>(value.as<std::string>()); return {}; } },
-                    { "Documentations", [this](const YAML::Node& value)->ValidationErrors { return ProcessDocumentationsNode(value); }},
+                    {
+                        "PurchaseUrl",
+                        [forRootFields](const YAML::Node& value, std::any& any)->ValidationErrors
+                        {
+                            ManifestLocalization* localization;
+                            if (forRootFields)
+                            {
+                                Manifest* manifest = std::any_cast<Manifest*>(any);
+                                localization = &(manifest->DefaultLocalization);
+                            }
+                            else
+                            {
+                                localization = std::any_cast<ManifestLocalization*>(any);
+                            }
+
+                            localization->Add<Localization::PurchaseUrl>(value.as<std::string>());
+                            return {};
+                        }
+                    },
+                    {
+                        "InstallationNotes",
+                        [forRootFields](const YAML::Node& value, std::any& any)->ValidationErrors
+                        {
+                            ManifestLocalization* localization;
+                            if (forRootFields)
+                            {
+                                Manifest* manifest = std::any_cast<Manifest*>(any);
+                                localization = &(manifest->DefaultLocalization);
+                            }
+                            else
+                            {
+                                localization = std::any_cast<ManifestLocalization*>(any);
+                            }
+
+                            localization->Add<Localization::InstallationNotes>(value.as<std::string>());
+                            return {};
+                        }
+                    },
+                    {
+                        "Documentations",
+                        [this, forRootFields](const YAML::Node& value, std::any& any)->ValidationErrors

```

Comment:
I thought the point of this was to get rid of the `this` capture.

Reply:
I can't get rid of it in all of them :(